kernel-scriptlets/inkmp-script

```diff
@@ -98,15 +95,15 @@ case $op in
 	# This is similar to the check for $system_map in rpm-script.
 	for __i in "${!modules[@]}"; do
 	    if [[ -e "${modules[$__i]}" ]]; then
-		unset -v modules["$__i"]
+		unset -v 'modules["$__i"]'
```

Comment:
This looks like a change in semantics

Reply:
That confused me too, but I checked it. Seems that it a removal of the element in the hash

Reply:
I was wondering about it, too, but I tested it and found it's indeed correct.

See https://www.shellcheck.net/wiki/SC2184 

I've seen lots of weird things in bash, but this is surely on the top ten list.

Reply:
reproducer:
```
> foo1=set
> echo $foo1
set
> foo=(a b c d)
> echo "${#foo[@]}"
4
> x=1
> touch foo1
> unset foo[$x]
> echo $foo1

> echo "${#foo[@]}"
4
> foo1=set
> unset 'foo[$x]'
> echo $foo1
set
> echo "${#foo[@]}"
3
```

Reply:
Relevant part in bash man page: "*When using a variable name with a subscript as an argument to a command, such as with unset, without using the word expansion syntax described above,  the  argument is subject to pathname expansion.  If pathname expansion is not desired, the argument should be quoted.*"